### PR TITLE
dockerfile fixes after hadolint scan

### PR DESCRIPTION
**Pull request by tkatila**
Signed-off-by: Tuomas Katila <tuomas.katila@intel.com>

### CODE REVIEW

**Comment by tkatila** [comment]
If/when the fixes are deemed ok, I'll "make dockerfiles" for the full effect.

File: build/docker/lib/toybox_build.docker [hunk 1]
```diff
@@ -3,7 +3,8 @@ ARG TOYBOX_SHA256="2bed6bb9edd5a249023103cf0402a835b0e53d10304a263f6f1e77a8aa49a
 
 ARG ROOT=/install_root
 
-RUN apt update && apt -y install musl musl-tools musl-dev
+RUN apt-get update && apt-get --no-install-recommends -y install musl=1.2.2-1 musl-tools=1.2.2-1 musl-dev=1.2.2-1
```
Comment (mythi) [inline build/docker/lib/toybox_build.docker:6]: I'd prefer we don't lock the versions even if the tool complains:
```suggestion
RUN apt-get update && apt-get --no-install-recommends -y install musl musl-tools musl-dev
```

**Comment by bart0sh** [comment]
@tkatila Dockerfile* and Dockerfile.in* are out of sync it seems. Here is what CI check reports:
 ```
Please commit result of "make dockerfiles" to make them in synch with the .in -files.
Error: Process completed with exit code 2.
```

File: build/docker/lib/default_build.docker [hunk 2]
```diff
@@ -1,8 +1,10 @@
 WORKDIR ${DIR}
 COPY . .
 
-RUN cd cmd/${CMD}; GO111MODULE=${GO111MODULE} CGO_ENABLED=0 go install "${BUILDFLAGS}"; cd - \N
-    && install -D /go/bin/${CMD} /install_root${EP}
+WORKDIR ${DIR}/cmd/${CMD}
+RUN GO111MODULE=${GO111MODULE} CGO_ENABLED=0 go install "${BUILDFLAGS}"
+WORKDIR ${DIR}
+RUN install -D /go/bin/${CMD} /install_root${EP}
```
Comment (bart0sh) [inline build/docker/lib/default_build.docker:7]: This would increase number of layers from one to 4. What's the benefit?

**Comment by tkatila** [comment]
> @tkatila Dockerfile* and Dockerfile.in* are out of sync it seems. Here is what CI check reports:
> 
> ```
> Please commit result of "make dockerfiles" to make them in synch with the .in -files.
> Error: Process completed with exit code 2.
> ```

Yes, I know. I wanted to propose these changes via the .in files first. And once the set of changes are agreed upon, I'd add the full .Dockerfile changeset.

File: build/docker/lib/default_build.docker (hunk 2 above)
Comment (tkatila) [reply build/docker/lib/default_build.docker:7]: Removal of a few hadolint warnings. But I don't think it's cost beneficial to add layers to remove a warning in this case. Thanks for noticing it!

File: build/docker/intel-deviceplugin-operator.Dockerfile [hunk 3]
```diff
@@ -25,7 +25,7 @@ ARG CMD=operator
 ## (see build-image.sh).
 ## 2) and the default FINAL_BASE is primarily used to build Redhat Certified Openshift Operator container images that must be UBI based.
 ## The RedHat build tool does not allow additional image build parameters.
-ARG FINAL_BASE=registry.access.redhat.com/ubi8-micro
+ARG FINAL_BASE=registry.access.redhat.com/ubi8-micro:8.6
```
Comment (mythi) [inline build/docker/intel-deviceplugin-operator.Dockerfile:28]: @chaitanya1731 is this selection OK for you?

File: build/docker/intel-fpga-initcontainer.Dockerfile [hunk 4]
```diff
@@ -75,7 +73,8 @@ RUN echo "{\n\
 ARG TOYBOX_VERSION="0.8.8"
 ARG TOYBOX_SHA256="2bed6bb9edd5a249023103cf0402a835b0e53d10304a263f6f1e77a8aa49a898"
 ARG ROOT=/install_root
-RUN apt update && apt -y install musl musl-tools musl-dev
+RUN apt-get update && apt-get --no-install-recommends -y install musl musl-tools musl-dev
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
```
Comment (bart0sh) [inline build/docker/intel-fpga-initcontainer.Dockerfile:77]: Just curious, what this is for?

Comment (tkatila) [reply build/docker/intel-fpga-initcontainer.Dockerfile:77]: https://github.com/hadolint/hadolint/wiki/DL4006

**Review by mythi — APPROVED** [review]
thanks!

**Comment by hj-johannes-lee** [comment]
Sorry, accidently I canceled your pr's ci test run..! :(

**Comment by tkatila** [comment]
> Sorry, accidently I canceled your pr's ci test run..! :(

It's probably fine. The previous version was tested and my last commit shouldn't change anything.

File: build/docker/lib/final_base.docker [hunk 5]
```diff
@@ -8,5 +8,5 @@
 ## (see build-image.sh).
 ## 2) and the default FINAL_BASE is primarily used to build Redhat Certified Openshift Operator container images that must be UBI based.
 ## The RedHat build tool does not allow additional image build parameters.
-ARG FINAL_BASE=registry.access.redhat.com/ubi8-micro
+ARG FINAL_BASE=registry.access.redhat.com/ubi8-micro:8.6
```
Comment (bart0sh) [inline build/docker/lib/final_base.docker]: Can you point me out to the explanation about this one?

**Review by bart0sh — DISMISSED** [review]
/lgtm

File: build/docker/lib/toybox_build.docker (hunk 1 above)
Comment (bart0sh) [reply build/docker/lib/toybox_build.docker]: I can still see versions for all packages? Are you going to apply @mythi's suggestion?

File: build/docker/lib/final_base.docker (hunk 5 above)
Comment (bart0sh) [reply build/docker/lib/final_base.docker]: nvm, was looking at the old change.

**Comment by bart0sh** [comment]
@tkatila can you squash your commits? I think it would be more readable when the same kind of changes are not spread on multiple commits.

File: build/docker/intel-deviceplugin-operator.Dockerfile (hunk 3 above)
Comment (chaitanya1731) [reply build/docker/intel-deviceplugin-operator.Dockerfile]: Looks good to me.

**Comment by tkatila** [comment]
> @tkatila can you squash your commits? I think it would be more readable when the same kind of changes are not spread on multiple commits.

Yes! Squashed and rebased.

**Review by mythi — APPROVED** [review]

**Pull request merged**